src/main/java/com/github/manolo8/darkbot/core/manager/HeroManager.java

```diff
@@ -199,10 +199,16 @@ public boolean attackMode(Npc target) {
         Config.ShipConfig config = this.main.config.GENERAL.OFFENSIVE;
 
         boolean otherConfig = target.npcInfo.extra.has(NpcExtra.OPPOSITE_CONFIG);
-        return setMode(
-                otherConfig ? ((config.CONFIG % 2) + 1) : config.CONFIG,
-                target.npcInfo.attackFormation != null ?
-                        target.npcInfo.attackFormation : config.FORMATION);
+
+        HeroAPI.Configuration newConfig = HeroAPI.Configuration.of(
+                otherConfig ? (config.getConfiguration().ordinal() % 2) + 1 : config.getConfiguration().ordinal()
```

Comment:
configuration should probably just have a "other" method that returns the next one, would simplify this